libraries/bot-applicationinsights/src/test/java/com/microsoft/bot/applicationinsights/BotTelemetryClientTests.java

```diff
@@ -0,0 +1,183 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.time.Duration;
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.verify;
+
+public class BotTelemetryClientTests {
+    final private BotTelemetryClient botTelemetryClient;
+    final private TelemetryChannel mockTelemetryChannel;
+
+    public BotTelemetryClientTests()
+    {
+        mockTelemetryChannel = Mockito.mock(TelemetryChannel.class);
+
+        TelemetryConfiguration telemetryConfiguration = new TelemetryConfiguration();
+        telemetryConfiguration.setInstrumentationKey("UNITTEST-INSTRUMENTATION-KEY");
+        telemetryConfiguration.setChannel(mockTelemetryChannel);
+        TelemetryClient telemetryClient = new TelemetryClient(telemetryConfiguration);
+
+        botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void NullTelemetryClientThrows() {
+        try {
+            new BotTelemetryClientImpl(null);
+        } catch (Exception e) {
+            Assert.assertEquals("IllegalArgumentException", e.getCause());
+        }
+    }
+
+    @Test
+    public void NonNullTelemetryClientSucceeds() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+
+        BotTelemetryClient botTelemetryClient = new BotTelemetryClientImpl(telemetryClient);
+    }
+
+    @Test
+    public void OverrideTest() {
+        TelemetryClient telemetryClient = new TelemetryClient();
+        MyBotTelemetryClient botTelemetryClient = new MyBotTelemetryClient(telemetryClient);
+    }
+
+    @Test
+    public void TrackAvailabilityTest()
+    {
+        Map<String, String> properties = new HashMap<>();
+        Map<String, Double> metrics = new HashMap<>();
+        properties.put("hello", "value");
+        metrics.put("metric", 0.6);
+
+        botTelemetryClient.trackAvailability(
+            "test",
+            OffsetDateTime.now(),
+            Duration.ofSeconds(1000), // TODO: use computer ticks
+            "run location",
+            true,
+            "message",
+            properties,
+            metrics);
+
+        TelemetryChannel telemetryClient = Mockito.mock(TelemetryChannel.class);
+        Mockito.doAnswer(invocation -> {
+            AvailabilityTelemetry eventName = invocation.getArgument(0);
+
+            Assert.assertEquals("test", eventName.getName());
+            Assert.assertEquals("value", eventName.getProperties().get("hello"));
+            Assert.assertEquals("0.6", eventName.getMetrics().get("metric"));
+
+            return null;
+        }).when(telemetryClient).send(Mockito.any(AvailabilityTelemetry.class));
+    }
+
+    @Test
+    public void TrackEventTest()
+    {
+        botTelemetryClient.trackEvent("test", new HashMap<String, String>() {{ put("hello", "value"); }}, new HashMap<String, Double>() {{ put("metric", 0.6); }});
```

Comment:
Can we separate this initializations as in the previous tests? I want to avoid doing a lot stuffs in a single line

Reply:
good idea, I like it